### DISCUSSION TITLE
Using k_thread's as a part of a class.

### DISCUSSION THREAD

**Discussion by dipakgmx** [zephyrproject-rtos/zephyr#50353] [Q&A] [answered]
I would like to implement an object with the thread object being a part of the class. Using the [K_THREAD_STACK_DEFINE](https://docs.zephyrproject.org/3.1.0/kernel/services/threads/index.html#group__thread__stack__api_1gac5368ce24fdeab3863b5c8dee2ebd955) results in errors. 

A MWE I have, is like this:

```
class myClass
{
public:
  myClass();

private:
  K_THREAD_STACK_DEFINE(m_stackArea, 1000);

};
```

But this results in errors during the build:

```
error: section attribute not allowed for 'm_stackArea'
K_THREAD_STACK_DEFINE(m_stackArea, 1000);
```

So, how does one go about including the thread as member variable within the class?

**Comment by jukkar** [comment, marked as answer]
Try to use `K_THREAD_STACK_MEMBER` that can be embedded within other data structures.

**Comment by dipakgmx** [comment]
I did come across this macro but wasn't sure, since it mentions something on the lines that this shouldn't be used.

`> This is now deprecated, as stacks defined in this way are not usable from user mode. Use K_KERNEL_STACK_MEMBER.`

So, now I did test using this macro and it works 👍, but am not sure if this is a long term solution, with the warning about this being deprecated. 

One more question, regarding this macro. The docs mention ` it is very important that any RAM preceding this member not be writable by threads else a stack overflow will lead to silent corruption`. How can I ensure this?
